scipy/sparse/tests/test_base.py

```diff
@@ -198,6 +211,9 @@ class _TestCommon:
     """test common functionality shared by all sparse formats"""
     math_dtypes = supported_dtypes
 
+    # TODO: Why this decorator is necessary when there is a module level
+    #       suppression, is completely puzzeling to me.
+    @sup_sparse_efficiency2.decorate
```

Comment:
I really do not understand this, this should happen inside the setup_module/teardown_module active suppress_warnings context. But I need an extra suppression here.